src/state/CallViewModel.ts

```diff
@@ -570,18 +635,34 @@ export class CallViewModel extends ViewModel {
       screenShares.length > 0
         ? ([of(screenShares.map((m) => m.vm)), this.spotlightSpeaker] as const)
         : ([
-            this.spotlightSpeaker.pipe(map((speaker) => [speaker!])),
+            this.spotlightSpeaker.pipe(
+              map((speaker) => (speaker ? [speaker] : [])),
+            ),
             this.spotlightSpeaker.pipe(
               switchMap((speaker) =>
-                speaker.local
-                  ? of(null)
-                  : this.localUserMedia.pipe(
-                      switchMap((vm) =>
-                        vm.alwaysShow.pipe(
-                          map((alwaysShow) => (alwaysShow ? vm : null)),
-                        ),
-                      ),
-                    ),
+                speaker
+                  ? speaker.local
+                    ? of(null)
+                    : this.mediaItems.pipe(
+                        switchMap((mediaItems) => {
+                          const localUserMedia = mediaItems.find(
+                            (m) => m.vm instanceof LocalUserMediaViewModel,
+                          ) as UserMedia | undefined;
+                          return (
+                            (
+                              localUserMedia?.vm as LocalUserMediaViewModel
+                            ).alwaysShow.pipe(
+                              map(
+                                (alwaysShow) =>
+                                  (alwaysShow
+                                    ? localUserMedia?.vm
+                                    : undefined) ?? null,
+                              ),
+                            ) ?? of(null)
+                          );
+                        }),
+                      )
+                  : of(null),
```

Comment:
I feel like this is at the edge of easy to grasp. It is very little and condense code for what its doing but we might want to consider:
 - returning a object with named properties
 - Be less LOC efficient and create some intermediate methods to:
 ```
 this.mediaItems.pipe(
  switchMap((mediaItems) => {
    const localUserMedia = mediaItems.find(
      (m) => m.vm instanceof LocalUserMediaViewModel,
    ) as UserMedia | undefined;
    return (
      (
        localUserMedia?.vm as LocalUserMediaViewModel
      ).alwaysShow.pipe(
        map(
          (alwaysShow) =>
            (alwaysShow
              ? localUserMedia?.vm
              : undefined) ?? null,
        ),
      ) ?? of(null)
    );
  }),
)
```

All of this could be wrapped in: `observePreferredPipItem(...)`
and we would just end up with:
```
([
  this.spotlightSpeaker.pipe(
    map((speaker) => (speaker ? [speaker] : [])),
  ),
  this.spotlightSpeaker.pipe(
    switchMap((speaker) =>
      speaker
        ? speaker.local
          ? of(null)
          : observePreferredPipItem(this)
    ),
  ),
] as const)
```

Reply:
> I feel like this is at the edge of easy to grasp. It is very little and condense code for what its doing but we might want to consider:
> 
> * returning a object with named properties

Done in 3d0bfaefe8743bd90c9b0267faf3d490219f6496.

Reply:
> > I feel like this is at the edge of easy to grasp. It is very little and condense code for what its doing but we might want to consider:
> > 
> > * returning a object with named properties
> 
> Done in [3d0bfae](https://github.com/element-hq/element-call/commit/3d0bfaefe8743bd90c9b0267faf3d490219f6496).

Having done this an obvious thing to do was to split `spotlightAndPip` directly into `spotlight` and `pip`. I don't know why we want them combined?

So, I did that in 33f398ddcf09f2b183abd43671a56a17098aca1b.

Then, I tried replacing some of the nested ternaries with if statements. I think this is more legible and would provide a nice place to add comments if appropriate.

This is in 7fdcbb32922c62c1413b5db47666fe0ec1718f86.

Reply:
the split of thwose two things is very nice!